### PR TITLE
Upgrade to java 21

### LINKED ISSUE #939: Upgrade to Java 21
**Description**

Zeebe is upgrading to Java 21. As we have a direct dependency on Zeebe we must also upgrade this project.

Ref: https://github.com/camunda/zeebe/issues/13716

### PR DESCRIPTION
**Pull request by remcowesterhoud**
## Description

Zeebe recently upgraded to Java 21 (https://github.com/camunda/zeebe/pull/14697).

To remain compatible with Zeebe this project must be upgraded.

## Related issues

closes #939

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to backport the fix

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually

Documentation:
* [ ] Javadoc has been written
* [ ] The documentation is updated

### CODE REVIEW

File: qa/embedded/pom.xml [hunk 1]
```diff
@@ -1,4 +1,20 @@
 <?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Copyright © 2021 camunda services GmbH (info@camunda.com)
```
Comment (Zelldon) [inline qa/embedded/pom.xml:3]: 🔧 We have now 2023 :D

**Review by Zelldon — APPROVED** [review]
🚀

**Comment by Zelldon** [comment]
Looks like build is failing.

File: qa/embedded/pom.xml (hunk 1 above)
Comment (remcowesterhoud) [reply qa/embedded/pom.xml:3]: I don't know how to configure the plugin to take the current hear 😄

**Comment by remcowesterhoud** [comment]
Not seeing this failure locally. I'll figure it out later when I'm not medic.

**Comment by remcowesterhoud** [comment]
bors merge

**Pull request merged**